### PR TITLE
Update poetry file to add FactoryBoy

### PR DESCRIPTION
**Pull request by stijn-janssens**

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ pytz = "^2024.1"
 boto3 = "^1.34.90"
 jinja2 = "^3.1.3"
 pydantic-settings = "^2.3.4"
+factory-boy = "^3.3.0"
```
Comment (andrii-ivaniuk) [inline backend/pyproject.toml:31]: Why did you add it here? `factory-boy` is added under `test` group, since it is required only for testing

Comment (stijn-janssens) [reply backend/pyproject.toml:31]: Sorry, forgot about that. Will change this PR to updating the README then to install this locally

**Review by andrii-ivaniuk — APPROVED** [review]
